openlibrary/plugins/openlibrary/js/SearchUtils.js

```diff
@@ -0,0 +1,136 @@
+import { removeURLParameter } from './Browser';
+import $ from 'jquery';
+
+/**
+ * Adds hidden input elements/modifes the action of the form to match the given search mode
+ * (Don't really understand what's happening, so just not gonna touch it too much)
+ * @param {JQuery} form the form we'll modify
+ * @param {String} searchMode
+ */
+export function addModeInputsToForm($form, searchMode) {
+    $('input[value=\'Protected DAISY\']').remove();
+    $('input[name=\'has_fulltext\']').remove();
+
+    let url = $form.attr('action');
+    url = removeURLParameter(url, 'm');
+    url = removeURLParameter(url, 'has_fulltext');
+    url = removeURLParameter(url, 'subject_facet');
+
+    if (searchMode !== 'everything') {
+        $form.append('<input type="hidden" name="m" value="edit"/>');
+        $form.append('<input type="hidden" name="has_fulltext" value="true"/>');
+        url += `${url.indexOf('?') > -1 ? '&' : '?'}m=edit&has_fulltext=true`;
+    }
+    if (searchMode === 'printdisabled') {
+        $form.append('<input type="hidden" name="subject_facet" value="Protected DAISY"/>');
+        url += `${url.indexOf('?') > -1 ? '&' : '?'}subject_facet=Protected DAISY`;
+    }
+
+    $form.attr('action', url);
+}
+
+
+/**
+ * @typedef {Object} PersistentValue.Options
+ * @property {any?} [default]
+ * @property {Function?} [initValidation] (str -> bool) validation to perform on intialization.
+ * @property {Function?} [writeTransformation] ((newVal, oldVal) -> val) function to call that
+ * transforms a value before save.
+ */
+
+/** String value that's persisted to localstorage */
+export class PersistentValue {
```

Comment:
I like the idea of having an abstraction for LocalStorage but I think it needs to be decoupled from the event emitting it does. Have you considered pulling in a library to do event emitting e.g. https://www.npmjs.com/package/event-emitter  or making this standalone and a mixin?

Reply:
See https://github.com/internetarchive/openlibrary/pull/2285#discussion_r321775857